XYZDrivers/src/main/java/com/xyzdrivers/repositories/UserRepo.java

```diff
@@ -0,0 +1,153 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package com.xyzdrivers.repositories;
+
+import com.xyzdrivers.models.User;
+import com.xyzdrivers.services.ColumnValuePair;
+import com.xyzdrivers.services.SQLService;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import javax.enterprise.context.RequestScoped;
+import javax.inject.Inject;
+
+/**
+ *
+ * @author Toneo
+ */
+@RequestScoped
+public class UserRepo extends Repo<User, String> {
+        
+    public UserRepo()
+    {
+        // The injection spec requires a parameterless constructor
+    }
+    
+    public UserRepo(SQLService sqlService)
+    {
+        this.sqlService = sqlService;
+    }
+    
+    @Override
+    public List<User> get() throws RepositoryException {
+        
+        List<Object[]> users;
+        try {
+            users = sqlService.retrieve("users");
+        } catch (SQLException ex) {
+            throw new RepositoryException("Retrieval failed.", ex);            
+        }
+        
+        return users.stream().map(parts -> 
+            new User((String)parts[0], 
+                     (String)parts[1], 
+                     (String)parts[2]))
+            .collect(Collectors.toList());
+        
+    }
+    
+    @Override
+    public User get(String id) throws RepositoryException {
+        
+        if (id == null || id.length() == 0) {
+            throw new IllegalArgumentException("Argument 'id' cannot be null or empty.");
+        }
+        
+        // TODO - specify primary key when calling on sqlService. SQLService is currently broken.
+        Stream<Object[]> allUsers;
+        try {
+            allUsers = sqlService.retrieve("users").stream();
+        } catch (SQLException ex) {
+            throw new RepositoryException("Retrieval failed.", ex);
+        }
+        
+        Object[] matches = allUsers
+                .filter(user -> ((String)user[0]).equals(id))
```

Comment:
Please do not use a different version of java than the spec. I personally do not have 1.8 ee on my machine